x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/CsvTests.java

```diff
@@ -224,7 +225,7 @@ public final void test() throws Throwable {
             assumeTrue("Test " + testName + " is not enabled", isEnabled(testName, Version.CURRENT));
 
             if (Build.current().isSnapshot()) {
-                assertThat(
+                assumeThat(
```

Comment:
I know this is not quite consistent with integration tests as-is, but I fear that this way we may accidentally skip tests e.g. because someone mistypes a capability name.

Additionally: integration tests need to work this way (skip test if capability not present) because otherwise bwc-tests don't work. The CSV tests (the non-integration version of them, i.e. the tests in this specific class) generally run only on main.

Suggestions:
- to allow for the workflow where a test is marked for skipping by marking it with a non-existent capability, we could restrict this to one specific capability name, like `awaitsFix` or `incompleteFeature` or `skipTest`, which is explicitly skipped via `assumeThat` - but we leave this `assertThat` in place.
- Or we could check that the capability starts with a given string, like `incompleteFeature`, to allow marking tests with e.g. `incompleteFeature:UnionTypes`.
- Or we could maintain a list of features that are being worked on, and just `assumeThat` the given tests has no required feature in that list. Additional benefit: removing the feature from that list automatically turns on all related csv tests.